basxconnect/mailer_integration/download_data.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, NamedTuple
+
+import basxconnect
+from basxconnect.core import models
+from basxconnect.mailer_integration.abstract.abstract_datasource import Datasource
+from basxconnect.mailer_integration.models import Interest, MailingPreferences
+
+
+class SynchronizationResult(NamedTuple):
+    new_persons: int
+    total_synchronized_persons: int
+
+
+def download_persons(datasource: Datasource) -> SynchronizationResult:
+    MailingPreferences.objects.all().delete()
+    Interest.objects.all().delete()
+
+    raw_persons = datasource.get_persons()
+    reader = datasource.person_reader()
+    datasource_tag = _get_or_create_tag(datasource.tag())
+    new_persons = 0
+    for raw_person in raw_persons:
+        matching_persons = list(
+            models.Person.objects.filter(
+                primary_email_address__email=reader.email_of(raw_person), deleted=False
+            ).all()
+        )
+        if len(matching_persons) == 0:
+            _save_person(datasource_tag, raw_person, reader)
+            new_persons += 1
+        else:
+            # if the downloaded email address already exists in our system, update the mailing preference of all persons
+            # that use this email address (as primary email address), without creating a new person in the database
+            for person in matching_persons:
+                _save_mailing_preferences(person, raw_person, reader)
+
+    return SynchronizationResult(new_persons, len(raw_persons))
+
+
+def _is_new_person(
+    person: Any,
+    reader: basxconnect.mailer_integration.abstract.abstract_person_reader.PersonReader,
+) -> bool:
+    return not models.Person.objects.filter(
+        primary_email_address__email=reader.email_of(person), deleted=False
+    ).exists()
+
+
+def _get_or_create_tag(tag: str) -> models.Term:
+    tags_category_id = models.Category.objects.get(slug="category").id
+    if not models.Term.objects.filter(term=tag, category_id=tags_category_id).exists():
+        models.Term(term=tag, category_id=tags_category_id).save()
```

Comment:
Nitpick: models.Term.objects.create(....) would be the modre Django-y approach, but according to the docs both approaches are equivalent.
In this case there is almost no difference but it is more about using the QuerySet-API when possible.